Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -536,6 +536,13 @@ void WebFrameProxy::setProcess(FrameProcess& process)
     m_frameProcess = process;
 }
 
+void WebFrameProxy::removeRemoteFrameProcessUse(WebFrameProxy& mainFrame)
+{
+    m_frameProcess = mainFrame.m_frameProcess;
```

Comment:
Why are you setting child frame processes to the main frame process? If we want to terminate child frame processes should we remove them from the browsing context group?

Reply:
That wasn't great.  We need to remove the child frames in the UI process which will remove the references to the FrameProcesses which will then clean up.